src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -1593,6 +1656,23 @@ int SketchObject::addGeometry(std::unique_ptr<Part::Geometry> newgeo, bool const
     return Geometry.getSize() - 1;
 }
 
+bool SketchObject::isClosedCurve(const Part::Geometry* geo)
+{
+    return (geo->is<Part::GeomCircle>()
+            || geo->is<Part::GeomEllipse>()
+            || (geo->is<Part::GeomBSplineCurve>()
+                && static_cast<const Part::GeomBSplineCurve*>(geo)->isPeriodic()));

```

Comment:
combining || and && in singular expression is hard to read. Contents of inner () should be extracted to its own variable:

```c++
const auto isPeriodicBSpline = geo->is<Part::GeomBSplineCurve>() && static_cast<const Part::GeomBSplineCurve*>(geo)->isPeriodic());
```

and then used in result.

I'm also not a fan of using static_casts here - we could use `as` helper inside class geometry.

Reply:
Again, `as` is new to me. Will take a look.

Reply:
It does not exist right now, but given the code I think that it would be quit useful to have as this contains a lot of casts.

Reply:
Sounds like a good candidate for an issue.

Coming to think of it, this may be exactly what dynamic cast does (and is probably what is being used in `geo->is`)/

Reply:
Yes, geo->is uses `isDerivedFrom` afair which is also base for `Base::freecad_dynamic_cast` - you can see how I implemented `getObject<T>` here: https://github.com/FreeCAD/FreeCAD/pull/17528/commits/3640318ce58d2bdd385329beb06d5ec0ec637417. This would be very similar.